### PR TITLE
handle the dot between prefix and key automatically

### PR DESCRIPTION
**Pull request by kevinkreiser**
fixe #27 

this also updates doc strings and README exmaples and since it is API breaking it adds version macros. let me know if i should remove the macros, i know we didnt really discuss them before.

### CODE REVIEW

**Comment by vthiery** [comment]
Thanks for the patch!

Regarding the macros, what's their use exactly?
I intend to cut the next release only when all the changes I attached to the `v2` milestone will be done.
I'll only update the version in the cmake file, tag and release, so unless there's a good reason, we should avoid setting these macros.

**Comment by kevinkreiser** [comment]
> what's their use exactly?

they allow downstream library maintainers to write protective code around API changes so that their library can work on whatever target platform they are being compiled. you might see something like this in downstream code:

```c++
// newer versions handle the dot automatically
#if STATSD_CLIENT_VERSION_MAJOR >= 2
StatsdClient client{ "127.0.0.1", 8080, "prefix", 20 };
// older versions require you to put the dot manually
#else
StatsdClient client{ "127.0.0.1", 8080, "prefix.", 20 };
#endif
```

this is probably less useful for a header only library where people are usually vendoring it rather than getting it as an installed package on their platform. but if the preferred way of getting this library is to install a package then the downstream library that depends on it might do like above to make sure whatever version they got from the package manager still worked in their code.

no problem if you think its not needed, i dont feel strongly about it, i can happily remove it!

File: tests/testStatsdClient.cpp
Comment (kevinkreiser) [inline tests/testStatsdClient.cpp:126]: this was just because i thought it would be useful to exorcise some bigger numbers some times

File: CMakeLists.txt
Comment (vthiery) [inline CMakeLists.txt:4]: Can we wait for a release to be prepared?

When we're ready to release v2, I'll open a PR and perform this change.

Comment (kevinkreiser) [reply CMakeLists.txt:4]: yep of course, no problem, i wasnt sure how you like to handle that! thanks

**Review by vthiery — APPROVED** [review]

**Pull request merged**
